src/aap_eda/services/validation.py

```diff
@@ -0,0 +1,97 @@
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import urllib.parse
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+from aap_eda.core import models
+from aap_eda.services.ruleset.activate_rulesets import ACTIVATION_PATH
+
+
+def validate_activation(
+    decision_environment_id: int,
+    rulebook_id: int,
+    user_id: int,
+    extra_var_id: int,
+) -> None:
+    _validate_awx_token(user_id)
+    _validate_decision_environment(decision_environment_id)
+    _validate_extra_var(extra_var_id)
+    _validate_rulebook(rulebook_id)
+    _validate_websocket()
+
+
+def _validate_awx_token(user_id) -> None:
+    tokens = models.AwxToken.objects.filter(user_id=user_id).count()
+    if tokens == 0:
+        raise ValidationError(_("No controller token specified"))
+    elif tokens > 1:
+        raise ValidationError(
+            _(
+                "More than one controller token found, "
+                "currently only 1 token is supported"
+            )
+        )
+
+
+def _validate_websocket() -> None:
+    ws_url = f"{settings.WEBSOCKET_BASE_URL}{ACTIVATION_PATH}"
+    parsed_url = urllib.parse.urlparse(ws_url)
+
+    if parsed_url.scheme not in ["ws", "wss"]:
+        raise ValidationError(
+            _(
+                "Connection Error: Invalid WebSocket URL scheme. "
+                "Scheme should be either 'ws' or 'wss'."
+            )
+        )
+    if not parsed_url.hostname:
+        raise ValidationError(
+            _(
+                "Connection Error: WebSocket URL must have a valid "
+                "host address."
+            )
+        )
+
+
+def _validate_decision_environment(decision_environment_id: int) -> None:
+    try:
+        decision_environment = models.DecisionEnvironment.objects.get(
+            id=decision_environment_id
+        )
+        if decision_environment.credential:
+            models.Credential.objects.get(
+                id=decision_environment.credential.id
+            )
+    except models.Credential.DoesNotExist:
+        raise ValidationError(_("No credential specified"))
+    except models.DecisionEnvironment.DoesNotExist:
+        raise ValidationError(_("No decision_environment specified"))
+
+
+def _validate_rulebook(rulebook_id: int) -> None:
+    try:
+        models.Rulebook.objects.get(id=rulebook_id)
+    except models.Rulebook.DoesNotExist:
+        raise ValidationError(_("No rulebook specified"))
+
+
+def _validate_extra_var(extra_var_id: int) -> None:
+    if not extra_var_id:
+        try:
+            models.ExtraVar.objects.get(id=extra_var_id)
+        except models.ExtraVar.DoesNotExist:
+            raise ValidationError(_("No extra_var specified"))
```

Comment:
The message is incorrect. Extra var id is optional, this function should check if the extra_var exists when it is provided.